### DISCUSSION TITLE
Exporting JSON Array of Objects as Excel

### DISCUSSION THREAD

**Discussion by moonswim** [exceljs/exceljs#1992] [Q&A]
I just found this library and reading through the doc which is very very very long and kind of disorienting i would like to know if it is possible and if so how tedious the process can be with this library to pass an array of objects in JSON and export it as an excel file.

Giving as each column header the key for each key value pair, and having the values inserted to each row.

Thank you in advance.

Cheers!

**Comment by ej-shafran** [comment]
You can create a workbook, add a worksheet, then use the `columns` field and `insertRows` option:

```js
const Excel = require("exceljs");

const workbook = new Excel.Workbook();
const worksheet = workbook.addWorksheet("Sheet1");
worksheet.columns = [
  { header: "ID", key: "id", width: 30 },
  { header: "Name", key: "name", width: 50 },
];
// insert from the second row onwards
worksheet.insertRows(2, [{ id: 1, name: "John Doe" }, { id: 2, name: "Jane Doe" }]);

// ... do whatever you want with the worksheet/workbook
```
